### PR TITLE
Tahoe-LAFS improvements

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
###### Things done
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I've deployed this update to a private grid and it runs fine.

### CODE REVIEW

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix [hunk 1]
```diff
@@ -48,8 +47,8 @@ pythonPackages.buildPythonApplication rec {
   '';
 
   checkPhase = ''
-    # TODO: broken with wheels
-    #${pythonPackages.python.interpreter} setup.py trial
+    # Still broken. ~ C.
+    #   trial allmydata
```
Comment (zimbatm) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: in that case you could leave the original comment

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix [hunk 2]
```diff
@@ -30,7 +30,6 @@ pythonPackages.buildPythonApplication rec {
     sed -i 's/"pycrypto.*"/"pycrypto"/' src/allmydata/_auto_deps.py
   '';
 
-  # Some tests want this + http://tahoe-lafs.org/source/tahoe-lafs/deps/tahoe-dep-sdists/mock-0.6.0.tar.bz2
   buildInputs = with pythonPackages; [ unzip numpy mock ];
```
Comment (zimbatm) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: why remove the useful comment ?

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix (hunk 1 above)
Comment (MostAwesomeDude) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: I have written a better comment; it's in the next patchset.

File: pkgs/tools/networking/p2p/tahoe-lafs/default.nix (hunk 2 above)
Comment (MostAwesomeDude) [inline pkgs/tools/networking/p2p/tahoe-lafs/default.nix]: Mostly 'cause it's totally wrong. A better comment is in the next patchset.

**Pull request merged**